### PR TITLE
add functionality to post new comment

### PR DESCRIPTION
**Pull request by matthew-sheehan**

### CODE REVIEW

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey 👋 I have reviewed your PR and it doesn't seem to be working. Please fix the functionality and resubmit when you're ready 😄

File: src/components/NewComment.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const [postSuccess, setPostSuccess] = useState(false)
+    const [isError, setIsError] = useState(false)
+    const [isLoading, setIsLoading] = useState(false)
+
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        setIsLoading(true)
+        postComment(newComment, article_id, user).then(postedComment => {
+            setComments(currentComments => {
+                return [postedComment, ...currentComments]
```
Comment (itskatherine) [inline src/components/NewComment.jsx:19]: Hi Matt, the second time you post a comment, everything breaks (though the API call goes through correctly) 
To save you some time debugging the root of the problem is that postedComments come back on the key of `comment`, whereas the currentComments aren't 🕵️ 🔍

File: src/components/NewComment.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                loading={isLoading}
+                size={20}
+                aria-label="Loading Spinner"
+                data-testid="loader"
+            /> : "Post Comment"}</button>}
+            {isError ? <p className="error-message">Something went wrong, please try again!</p> : null}
+        </form>
+    </div>
+}
+
```
Comment (itskatherine) [inline src/components/NewComment.jsx:52]: I am currently able to spam the button which submits a comment and post loads while the loading icon is whirling and post loads of the same comment. Can you add some functionality to stop the user being able to do this please 🚫

File: src/utils/apis.js [hunk 3]
```diff
@@ -17,4 +17,15 @@ export const getTopics = () => {
...
         .then(res => res.data)
+}
+
+export const postComment = (body, article_id, username) => {
+    console.log(body, article_id, username, "<<<< post comment params")
+    return api.post(`/articles/${article_id}/comments`, {
+        body: body,
+        username: username
+      }).then(res => {
+        console.log(res)
```
Comment (itskatherine) [inline src/utils/apis.js]: `console.log` begone 🧙

File: src/utils/apis.js [hunk 4]
```diff
@@ -17,4 +17,15 @@ export const getTopics = () => {
...
+        return res.data
+      })
```
Comment (itskatherine) [inline src/utils/apis.js]: I would take the comment of it's key here if I were you

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Matt, great implementation of a pop up and loading state 🙌 
However some points to look over below, some minor issues that are breaking your app - once you've fixed them could you please resubmit this ticket 🎫 

Also something to consider in the future (I won't ask you to do it now) is how this UI reacts to mobile screens, as currently it messes up on mobile screens, though looks great on desktop! 📱

**Review by corkesw — COMMENTED** [review]
Great work on the fixes 🙌 This looks good to merge.

**Pull request merged**